DUI3-DX/Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Speckle.Connectors.Utils.Builders;
+using Speckle.Core.Api;
+using Speckle.Core.Credentials;
+using Speckle.Core.Logging;
+using Speckle.Core.Models;
+using Speckle.Core.Transports;
+
+namespace Speckle.Connectors.Utils.Operations;
+
+public sealed class ReceiveOperation
+{
+  private readonly IHostObjectBuilder _hostObjectBuilder;
+
+  public ReceiveOperation(IHostObjectBuilder hostObjectBuilder)
+  {
+    _hostObjectBuilder = hostObjectBuilder;
+  }
+
+  public async Task<IEnumerable<string>> Execute(
+    string accountId, // POC: all these string arguments exists in ModelCard but not sure to pass this dependency here, TBD!
+    string projectId,
+    string projectName,
+    string modelName,
+    string versionId,
+    CancellationTokenSource cts,
```

Comment:
Question: Do we need to be passing the token source around?
the general pattern for cancellation is to keep a reference to the CancellationTokenSource in a single place (the object that "owns" the token source), and to pass only the `CancellationToken` into methods as the last parameter.

I'm curious if we are breaking this pattern intentionally?

Reply:
see https://learn.microsoft.com/en-us/dotnet/standard/threading/cancellation-in-managed-threads and https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource?view=net-8.0